src/alchemlyb/workflows/abfe.py

```diff
@@ -82,6 +85,10 @@ def __init__(
             f"{software}"
         )
         reg_exp = "**/" + prefix + "*" + suffix
+        if '*' in dir:
+            warnings.warn(f"A real directory is expected in `dir`={dir}, wildcard expressions should be supplied to `prefex`.")
```

Comment:
```suggestion
            warnings.warn(f"A real directory is expected in `dir`={dir}, wildcard expressions should be supplied to `prefix`.")
```

Reply:
If we find a `*`, will we then fail with ValueError on the next line?

Reply:
Depends on whether this is a directory or not if there is a directory that is literally named as `folder_*`, then it will be fine.